### DISCUSSION TITLE
Update translation cache forcefully for a specific page

### DISCUSSION THREAD

**Discussion by iG8R** [immersive-translate/immersive-translate#1417] [Help]
### 概述 | Summary

Is there any way to forcefully update translation cache in immersive-translate for a specific page without deleting the entire cache via `Options -> About` `Cache size: Calculating... [Clear cache]`?

**Comment by theowenyoung** [comment]
Currently, if you use the Greasemonkey script, you can clear the indexDB of the current website in the developer tools to clear the cache of the current website. As for addons, there is currently no way. You can temporarily disable the cache to use the latest results

**Comment by iG8R** [comment]
Thanks a lot!
I use the Firefox addon, very often I encounter issues when Gemini incorrectly distributes translated paragraphs, combining several of them in one place and leaving others empty, which forces me to change the `Maximum number of paragraphs per request` from 1 to 5 to avoid confusion with the translation on the page, but due to the cache it becomes quite a tedious task because of all these manipulations with turning the cache on and off.
Could you please add some methods to edit the cache by the user?

**Reply by theowenyoung** [reply]
Unfortunately, this is not easy because the cache in the current addon is cached at the paragraph level. It has nothing to do with the website, only in this way it can greatly reduce the need for repeated translation of paragraphs.

If you encounter similar issues, perhaps you need to change the maximum number of paragraphs to 1 instead of 5. This will avoid most AI problems.

**Reply by iG8R** [reply]
That's exactly what I do - I change the max value to 1, which causes a drastic decrease in translation speed and error `429: { "error": { "code": 429, "message": "Resource has been exhausted (e.g. check quota).", "status": "RESOURCE_EXHAUSTED". } } } `, which forces me to refresh the translation several times when I encounter that the position of the translated text is confused. But this doesn't happen with every site, so by default I set it to 10, but you can't predict in advance which page will have translation distribution issues.
That's why it turns out that the default value is 10, but when a page is already translated with this value and the position of the translated text is messed up, I have to either clear the cache completely, which is sometimes quite painful, because the cache size can be 100MB or more, or disable it to re-translate the page with a value of 1, causing the correct cache to not persist because it is disabled.
